pkg/agent/cniserver/interface_configuration_linux.go

```diff
@@ -250,7 +251,17 @@ func (ic *ifConfigurator) configureContainerLinkVeth(
 	containerIface := &current.Interface{Name: containerIfaceName, Sandbox: containerNetNS}
 	result.Interfaces = []*current.Interface{hostIface, containerIface}
 
-	podMAC := util.GenerateRandomMAC()
+	var podMAC net.HardwareAddr
+	if mac != "" {
```

Comment:
I wonder if the function should take a `mac net.HardwareAddr` parameter instead, and parsing / checking the user-provided MAC address should happen earlier, in the controller?

Reply:
@KMAnju-2021 
could you check if validation for provided mac address can be done in configurePodSecondaryNetwork's validation method !

Reply:
I don't think the change has much value then? The value comes from handling user configuration errors as early as possible, in the controller logic, so we can differentiate between user errors and transient system errors.

Reply:
done